packages/openai/src/responses/openai-responses-language-model.ts

```diff
@@ -303,13 +340,32 @@ export class OpenAIResponsesLanguageModel implements LanguageModelV2 {
             });
 
             for (const annotation of contentPart.annotations) {
-              content.push({
-                type: 'source',
-                sourceType: 'url',
-                id: this.config.generateId?.() ?? generateId(),
-                url: annotation.url,
-                title: annotation.title,
-              });
+              switch (annotation.type) {
+                case 'url_citation':
+                  content.push({
+                    type: 'source',
+                    sourceType: 'url',
+                    id: this.config.generateId?.() ?? generateId(),
+                    url: annotation.url,
+                    title: annotation.title,
+                  });
+                  break;
+                case 'file_citation':
+                  content.push({
+                    type: 'source',
+                    sourceType: 'document',
+                    id: this.config.generateId?.() ?? generateId(),
+                    mediaType: this.inferMediaType(annotation.filename),
```

Comment:
why do we need this? is it us enforcing the media type on the source? if so, what is the value and should we remove it